client/src/components/Trade/Trade.jsx

```diff
@@ -26,35 +26,80 @@ class Trade extends React.Component {
 
   handleInputChange(event) {
     const target = event.target;
-    const value = Math.floor(target.value);
+    const value = Math.floor(Math.abs(target.value));
     const name = target.name;
 
-    this.setState({
-      [name]: value
-    });
+    const currentUser = this.props.user;
+    const shares = this.state.shares;
+    const stockSymbol = this.props.stockSelected.symbol;
+    const stockOwned = currentUser !== undefined ? currentUser.userPortfolio.filter(stock => {
+      return stock.ticker_symbol === stockSymbol;
+    }) : 0;
+    const price = this.props.stockSelected.price;
+    const cashBalance = (this.props.user.cashBalance/100).toFixed(2);
+    const action = this.props.action;
+
+    if (tradeValidation(shares, stockOwned, price, cashBalance, action)) {
+      this.setState({
+        [name]: value,
```

Comment:
This allows you to declare a variable as a key name, correct?

Reply:
Yep!